src/plugins/cordova-plugin-camera/app-host-handlers.js

```diff
@@ -5,9 +5,14 @@ module.exports = function (messages) {
         Camera: {
             takePicture: function (success, fail, args) {
                 messages.call('takePicture', args).then(function (result) {
-                    // 'result' should be {data: <ArrayBuffer>, type: <mimeType>}, from which we'll create a blob
-                    var blob = new Blob([result.data], {type: result.type});
-                    success(URL.createObjectURL(blob));
+                    if (args && args[1] === 0) {
```

Comment:
You are right, there is no work to be done here for the NATIVE_URI destination type. For both FILE_URI and NATIVE_URI we will return a HTTP URL which will then be displayed in APP_HOST accordingly. The only exception is the DATA_URL for which we need to return the base64 encoded image as a string.